src/main/java/Parser.java

```diff
@@ -0,0 +1,67 @@
+public class Parser {
+    private String[] information;
+    private String date;
+    private Command command;
+    private String description;
+    private int taskIndex;
+
+    public Parser(String input) {
+        try {
+            this.information = input.split(" ", 2);
+            this.command = Command.valueOf(this.information[0].toUpperCase());
+
+            switch (this.command) {
+                case FIND:
+                    this.description = this.information[1];
+                    break;
+                case DELETE:
+                case DONE:
+                    this.taskIndex = Integer.parseInt(this.information[1]);
+                    break;
+                case TODO:
+                case DEADLINE:
+                case EVENT:
+                    String[] descriptionAndTime = information.length < 2 ? null : information[1].split("/", 2);
+                    this.description = descriptionAndTime == null ? "DESCRIPTION.ERROR" : descriptionAndTime[0];
+                    this.date = descriptionAndTime == null || descriptionAndTime.length < 2 ? "DATE.ERROR"
```

Comment:
Perhaps I could suggest declaring a static final variable to define what 2 is supposed to be here, I believe it's to ensure that there are sufficient arguments but maybe you could name it something like MIN_VARS :)

https://nus-cs2103-ay2021s2.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-magic-numbers